optimizely/marketing-website
# assemble upgrades

This PR is in response to localization subfolders being put on hold. Unfortunately, there was a lot of work done that would have been too difficult to backtrack so I had to branch off of https://github.com/optimizely/marketing-website/pull/1020. 

## Noticeable changes
- `MD_page_content: true` is no longer needed for partner pages. Markdown is being parsed by the markdown helper because even with localization this content does not need to be translated.
- With `node 0.12.4` watch task call stack errors will not happen. @brianlucas it is recommended to upgrade our production/staging install script to use `node 0.12.4`.
- error tracking in the `create-file-data` plugin is improved
- running ui-tests has changed because there are now assemble "server side" tests in the `mochaTest` config. To run individual ui tests with server running `grunt ui-test --target=<your test name>. To run all the tests for ui `grunt ui-test`

## To Test
- run `grunt test`, or `grunt mochaTest:assemble` & `grunt ui-test`
- check partners and resources pages to make sure all navs work
- check dialogs, forms and images
- ensure `SL_swap` etc are still in the markup
- use `node 0.12.4` then `rmall` and then fire up the server and save a bunch of different assemble hbs and/or yml files to see if the watch task continues to work.
